tests/test_compile_kwargs.py

```diff
@@ -0,0 +1,555 @@
+
+    est = KerasClassifier(
+        model=get_model,
+        loss=MyLoss,
+        loss__param1=[Foo, Foo],
+        loss__param1__foo_kwarg=1,
+        loss__param1__0__foo_kwarg=2,
+    )
+    est.fit(X, y)
+    assert est.model_.loss.param1[0].foo_kwarg == 2
+    assert est.model_.loss.param1[1].foo_kwarg == 1
+
+
+@pytest.mark.parametrize(
+    "loss", (losses_module.BinaryCrossentropy, "BinaryCrossentropy",),
+)
+@pytest.mark.parametrize("n_outputs_", (1, 2))
+def test_loss(loss, n_outputs_):
+    """Tests compiling of single loss
+    using routed parameters.
+    """
+
+    X, y = make_classification()
+    y = np.column_stack([y for _ in range(n_outputs_)]).squeeze()
+
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss=loss,
+        loss__name="custom_name",
+    )
+    est.fit(X, y)
+    assert str(loss) in str(est.model_.loss) or isinstance(
+        est.model_.loss, loss
+    )
+
+
+def test_loss_invalid_string():
+    """Tests that a ValueError is raised when an unknown
+    string is passed as a loss.
+    """
+
+    X, y = make_classification()
+
+    loss = "binary_crossentropr"  # binary_crossentropr is not a loss
+
+    est = KerasClassifier(
+        model=get_model, num_hidden=20, optimizer="sgd", loss=loss,
+    )
+    with pytest.raises(ValueError, match="Unknown loss function"):
+        est.fit(X, y)
+
+
+def test_loss_uncompilable():
+    """Tests that a TypeError is raised when a loss
+    that is not compilable is passed routed parameters.
+    """
+
+    X, y = make_classification()
+
+    loss = losses_module.binary_crossentropy
+
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss=loss,
+        loss__from_logits=True,
+    )
+    expected_param = {"from_logits": True}
+    with pytest.raises(
+        TypeError,
+        match=f'TypeError: "{str(loss)}" object of type "{type(loss)}"'
+        "does not accept parameters because it's not a class."
+        f' However, it received parameters "{expected_param}"',
+    ):
+        est.fit(X, y)
+
+
+@pytest.mark.parametrize(
+    "loss", (losses_module.BinaryCrossentropy, "BinaryCrossentropy",),
+)
+@pytest.mark.parametrize("n_outputs_", (1, 2))
+def test_loss_routed_params_iterable(loss, n_outputs_):
+    """Tests compiling of loss when it is
+    given as an iterable of losses
+    mapping to outputs.
+    """
+
+    X, y = make_classification()
+    y = np.column_stack([y for _ in range(n_outputs_)]).squeeze()
+
+    # Test iterable with global routed param
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss=[loss],
+        loss__from_logits=True,  # default is False
+    )
+    est.fit(X, y)
+    assert est.model_.loss[0].from_logits
+
+    # Test iterable with index-based routed param
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
```

Comment:
Maybe `num_hidden=20` would be a default value for `get_model` (`num_hidden=20` is scattered all over these tests).

Reply:
Thanks. No reason to have these. Test is clearer without. I just deleted all of these lines.